Neos.EventSourcedContentRepository/Classes/Domain/Context/NodeAggregate/Property/Normalizers/CustomArrayDenormalizer.php

```diff
@@ -0,0 +1,91 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\EventSourcedContentRepository\Domain\Context\NodeAggregate\Property\Normalizers;
+
+use Symfony\Component\Serializer\Exception\BadMethodCallException;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\SerializerAwareInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Understands array of objects in the array<Type> notation used by Flow.
+ *
+ * @final
+ */
+class CustomArrayDenormalizer implements ContextAwareDenormalizerInterface, SerializerAwareInterface, CacheableSupportsMethodInterface
+{
+    /**
+     * @var SerializerInterface|DenormalizerInterface
+     */
+    private $serializer;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotNormalizableValueException
+     */
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        if (null === $this->serializer) {
+            throw new BadMethodCallException('Please set a serializer before calling denormalize()!');
```

Comment:
No timestamp on the exception (and all the others) – don't we adhere to the general CGL in this part of the code?